phpBB/phpbb/members/controller/online.php

```diff
@@ -0,0 +1,618 @@
+<?php
+/**
+ *
+ * This file is part of the phpBB Forum Software package.
+ *
+ * @copyright (c) phpBB Limited <https://www.phpbb.com>
+ * @license GNU General Public License, version 2 (GPL-2.0)
+ *
+ * For full copyright and license information, please see
+ * the docs/CREDITS.txt file.
+ *
+ */
+
+namespace phpbb\members\controller;
+
+use \phpbb\auth\auth;
+use \phpbb\config\config;
+use \phpbb\db\driver\driver_interface;
+use \phpbb\event\dispatcher;
+use \phpbb\group\helper as group_helper;
+use \phpbb\controller\helper;
+use \phpbb\language\language;
+use \phpbb\pagination;
+use \phpbb\request\request;
+use \phpbb\routing\router;
+use \phpbb\template\template;
+use \phpbb\user;
+
+class online
+{
+	/**
+	 * @var auth
+	 */
+	protected $auth;
+
+	/**
+	 * @var config
+	 */
+	protected $config;
+
+	/**
+	 * @var driver_interface
+	 */
+	protected $db;
+
+	/**
+	 * @var dispatcher
+	 */
+	protected $dispatcher;
+
+	/**
+	 * @var group_helper
+	 */
+	protected $group_helper;
+
+	/**
+	 * @var helper
+	 */
+	protected $helper;
+
+	/**
+	 * @var language
+	 */
+	protected $language;
+
+	/**
+	 * @var pagination
+	 */
+	protected $pagination;
+
+	/**
+	 * @var request
+	 */
+	protected $request;
+
+	/**
+	 * @var router
+	 */
+	protected $router;
+
+	/**
+	 * @var template
+	 */
+	protected $template;
+
+	/**
+	 * @var user
+	 */
+	protected $user;
+
+	public function __construct(auth $auth, config $config, driver_interface $db, dispatcher $dispatcher, group_helper $group_helper, helper $helper, language $language, pagination $pagination, request $request, router $router, template $template, user $user)
+	{
+		$this->auth				= $auth;
+		$this->config			= $config;
+		$this->db				= $db;
+		$this->dispatcher		= $dispatcher;
+		$this->group_helper		= $group_helper;
+		$this->helper			= $helper;
+		$this->language			= $language;
+		$this->pagination		= $pagination;
+		$this->request			= $request;
+		$this->router			= $router;
+		$this->template			= $template;
+		$this->user				= $user;
+	}
+
+	/**
+	 * Controller for /online route
+	 *
+	 * @return \Symfony\Component\HttpFoundation\Response a Symfony response object
+	 */
+	public function handle()
+	{
+		global $phpbb_root_path, $phpEx;
+
+		// Display a listing of board admins, moderators
+		if (!function_exists('display_user_activity'))
+		{
+			include($phpbb_root_path . 'includes/functions_display.' . $phpEx);
```

Comment:
Ideally the needed functions would be refactored into some kind of helper to prevent us from having to include function files.

Reply:
Yes, this is mostly a copy/paste to controllers to be able to use routes, I haven't rewriten much